activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,16 @@
+*   Add `ActiveRecord::Relation#with_annotation` for adding SQL comments to its queries.
+
+    For example:
+
+    ```
+    Post.where(id: 123).with_annotation("this is a comment").to_sql
+    # SELECT "posts".* FROM "posts" WHERE "posts"."id" = 123 /* this is a comment */
+    ```
+
+    This can be useful in instrumentation or other analysis of issued queries.
+
+    *Matt Yoho*
+
```

Comment:
The `with_` feels mismatched with other relation builder methods, suggesting e.g. an annotation that applies within the scope of a block. Analog would be `with_conditions` vs `where` or `with_limit` vs `limit`. Would compare with `Post.where(id: 123).annotate("this is a comment")` or similar.